policyengine_us/variables/gov/states/mi/tax/income/mi_household_resources.py

```diff
@@ -8,11 +8,50 @@ class mi_household_resources(Variable):
     unit = USD
     definition_period = YEAR
     defined_for = StateCode.MI
-    reference = "https://law.justia.com/codes/michigan/2022/chapter-206/statute-act-281-of-1967/division-281-1967-1/division-281-1967-1-9/section-206-508/"
+    reference = (
+        "https://law.justia.com/codes/michigan/2022/chapter-206/"
+        "statute-act-281-of-1967/division-281-1967-1/division-281-1967-1-9/"
+        "section-206-508/",
+        "https://www.michigan.gov/taxes/-/media/Project/Websites/taxes/"
+        "Forms/IIT/TY2024/BOOK_MI-1040CR-7.pdf",
+    )
 
     def formula(tax_unit, period, parameters):
         p = parameters(period).gov.states.mi.tax.income
-        household_resources = add(tax_unit, period, p.household_resources)
+        # Per MI-1040CR form instructions, only certain income sources must be
+        # floored at 0 if negative. Capital gains have special loss limitation.
+        income_sources = p.household_resources
+
+        # MI-1040CR Line 16: "Net business income (including net farm income).
+        # If negative, enter 0"
+        business_income_sources = [
+            "farm_income",
+            "self_employment_income",
+            "partnership_s_corp_income",
+        ]
+        # MI-1040CR Line 17: "Net royalty or rent income. If negative, enter 0"
+        rent_income_sources = [
+            "rental_income",
+            "farm_rent_income",
+        ]
+        # Sources that must be floored at 0 per form instructions
+        floored_sources = business_income_sources + rent_income_sources
```

Comment:
Those are already in the list of household resources, we are double counting those now